BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -1707,68 +1709,59 @@ unsafe public override IProjectiveTransformation GetIntrinsics(string channelNam
+                    streamType = Stream.Color;
                     break;
                 case ChannelNames.ZImage:
-                case ChannelNames.Left:
-                case ChannelNames.Right:
                 case ChannelNames.Distance:
-                default:
                     sensorName = SensorNames.Stereo;
+                    streamType = Stream.Depth;
                     refResolution = DepthResolution;
+                    refFPS = DepthFPS;
                     break;
-            }
-
-            StreamProfileList list = GetSensor(sensorName).StreamProfiles;
-
-            foreach(VideoStreamProfile profile in list)
-            {
-                Point2i res = new Point2i(profile.Width, profile.Height);
-                if (res == refResolution)
-                    return profile;
-            }
-
-            throw new ArgumentException(string.Format("StreamProfile for channel '{0}' with resolution {1}x{2} notavailable", channelName, refResolution.X, refResolution.Y));
-        }
-
-        private StreamProfile GetProfileFromCapturedFrames(string channelName)
-        {
-            Frame frame;
-
-            switch (channelName)
-            {
-                case ChannelNames.Color:
-                    frame = _currentColorFrame;
-                    break;
-
-                case ChannelNames.ZImage:
-                    frame = _currentDepthFrame;
-                    break;
-
                 case ChannelNames.Left:
-                    frame = _currentLeftFrame;
+                    sensorName = SensorNames.Stereo;
+                    streamType = Stream.Infrared;
+                    refResolution = DepthResolution;
+                    refFPS = DepthFPS;
+                    index = 1;
                     break;
-
                 case ChannelNames.Right:
-                    frame = _currentRightFrame;
+                    sensorName = SensorNames.Stereo;
+                    streamType = Stream.Infrared;
+                    refResolution = DepthResolution;
+                    refFPS = DepthFPS;
+                    index = 2;
                     break;
-
                 default:
                     string msg = string.Format("RealSense2: stream profile for channel {0} not available", channelName);
                     log.Error(msg);
                     throw new ArgumentException(msg, nameof(channelName));
             }
 
-            return frame.Profile;
+            Sensor sensor = GetSensor(sensorName);
+
+            return sensor.VideoStreamProfiles
+                .Where(p => p.Stream == streamType)
+                .Where(p => p.Framerate == refFPS)
+                .Where(p => p.Width == refResolution.X && p.Height == refResolution.Y)
+                .Where(p => p.Index == index)
+                .First();
+
+            throw new ArgumentException(string.Format("StreamProfile for channel '{0}' with resolution {1}x{2} notavailable", channelName, refResolution.X, refResolution.Y));
```

Comment:
Is this line reachable?

Reply:
No, leftover code from how it worked before.